frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,479 @@
-import React from "react";
+import React, { useMemo, useRef, useState } from "react";
+import {
+    Table,
+    Icon,
+    Modal,
+    ModalRef,
+    ModalHeading,
+} from "@trussworks/react-uswds";
 
-import { DetailItem } from "../DetailItem/DetailItem";
 import { ReceiverData } from "../../config/endpoints/messageTracker";
-import { formattedDateFromTimestamp } from "../../utils/DateTimeUtils";
-
-import { QualityFilters } from "./QualityFilters";
+import { parseFileLocation } from "../../utils/misc";
 
 type MessageReceiverProps = {
-    receiverDetails: ReceiverData[] | undefined;
+    receiverDetails: ReceiverData[];
+};
+
+interface NormalizedReceiverData {
+    Name: string;
+    Service: string;
+    Date: string;
+    ReportId: string;
+    Main: "Batch" | "Process" | "Ready";
+    Sub: string;
+    FileName: string;
+    TransportResults: string;
+}
+
+interface MessageReceiversRowProps {
+    receiver: NormalizedReceiverData;
+    activeColumn: string;
+    activeColumnSortOrder: string;
+    setModalText: ({ title, body }: { title: string; body: string }) => void;
+    modalRef: React.RefObject<ModalRef>;
+}
+
+interface MessageReceiversColProps {
+    columnHeaderTitle: string;
+    activeColumn: string;
+    setActiveColumn: (colTitle: string) => void;
+    activeColumnSortOrder: string;
+    setActiveColumnSortOrder: (sortOrder: string) => void;
+    filterIcon: JSX.Element;
+    rowSpan: number;
+}
+
+const dateTimeFormatter = new Intl.DateTimeFormat("en-US", {
+    month: "2-digit",
+    day: "2-digit",
+    year: "numeric",
+    hour: "2-digit",
+    minute: "2-digit",
+    second: "2-digit",
+});
+
+const ColumnDataEnum = {
+    Name: "Name",
+    Service: "Service",
+    Date: "Date",
+    ReportId: "Report Id",
+    Main: "Main",
+    Sub: "Sub",
+    FileName: "File Name",
+    TransportResults: "Transport Results",
+};
+
+const FilterOptionsEnum = {
```

Comment:
Out of curiosity, why are these consts rather than real enums?

Reply:
I was getting some strange TS errors about using them as a `key` in places, and from what I see in TS, the TS `enum` just compiles down to an object anyways. I can swap these back to `enum` and debug.

Reply:
No worries -- these could also be problematic if they're needed at compile-time versus runtime.  If that's the case, then keeping at consts is fine, but the `Enum` suffix is just a bit unnecessary :)